### PR TITLE
Preview: Initial loading state

### PR DESCRIPTION
**Pull request by marekhrabe**
This PR takes the loader styling from `EditorPreview` and brings it to general `WebPreviewContent` component so everybody can use it without the need for extra CSS.

It also adds the preview message to "View Site" section (/view/:slug).

The loading message can only be seen for the initial load of the preview. All subsequent page loads (that user triggers by clicking links in preview) have no loading state, yet.

This PR also adds `pointer-events: none` to the preview iframe that is hidden so you cannot interact with invisible iframe.

**Test:**

- go to https://calypso.live/view/?branch=update/view-site-loading-state and pick your favorite site
- instead of the grey screen, see our new loading message

**Visual:**

screen shot 2017-07-28 at 12 37 33

**For designers:**

The spinner circle above the text was there all the time but it was broken by CSS, thus invisible. Should we keep it there? I'd personally remove it and keep just the text content there (that's what you can currently see in post-post) and combine this with the `SpinnerLine` from #16675.

**Todo:**
- [x] based on design review, maybe remove the circular spinner

### CODE REVIEW

File: client/my-sites/preview/main.jsx [hunk 1]
```diff
@@ -119,6 +119,11 @@ class PreviewMain extends React.Component {
 					showClose={ false }
 					previewUrl={ this.state.previewUrl }
 					externalUrl={ this.state.externalUrl }
+					loadingMessage={
+						this.props.translate( '{{strong}}One moment please…{{/strong}} loading your site.',
```
Comment (a8ci18n) [inline client/my-sites/preview/main.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( '{{strong}}One moment please…{{/strong}} loading your new post.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by shaunandrews** [comment]
> The spinner circle above the text was there all the time but it was broken by CSS, thus invisible. Should we keep it there?

Ditch it. The general rule in Calypso is to avoid loading spinners. That being said, I wonder if it would be nice to do something like this to at least tell people that "something" is happening:

dot dot dot

--

Got this in Safari:

screen shot 2017-07-28 at 3 03 17 pm

File: client/components/web-preview/style.scss [hunk 2]
```diff
@@ -297,12 +299,30 @@ a.web-preview__external.button {
...
+	color: $gray-text-min;
 	font-weight: 300;
 	text-align: center;
 	display: block;
-	margin: 48px 0;
-	height: 100%;
+	font-size: 16px;
+}
+
+.web-preview__loading-message strong {
```
Comment (spen) [inline client/components/web-preview/style.scss]: I don't think it matters too much in a case like this personally, but do we usually prefer to add a class and style the class (for the `strong` that is)?

Feel free to ignore this

File: client/components/web-preview/style.scss [hunk 3]
```diff
@@ -297,12 +299,30 @@ a.web-preview__external.button {
...
+	color: #537994;
```
Comment (spen) [inline client/components/web-preview/style.scss]: Should the colour be taken from a set of colours?
From a search it looks like `$gray-text-min` would do the trick (`stylesheets/shared/_colors.scss`, line 15)

File: client/my-sites/preview/main.jsx (hunk 1 above)
Comment (spen) [inline client/my-sites/preview/main.jsx]: I know nothing about how `a8ci18n` works, but it feels like we could have a set of keywords ('post', 'site', 'project' etc.) that are recognised as not being interchangeable - the bot would see that the difference is these 2 words and know not to offer a suggestion.
Though, the above is still useful even just for making sure the formatting of the sentence is the same...

**Comment by spen** [comment]
Also, just my opinion but I find:
"One moment please..."
"Loading your site"

to be a little funky - to me the ellipsis makes more sense on the Loading you site... - to me that's whats 'pending'.
It feels to me as though the wait of the `...` is capped by the following sentence.

"One moment please."
"Loading your site..."

Looks better in my eyes.

I'd probably just scratch this down to me being a bit weird, but thought I'd mention it any way `¯\_(ツ)_/¯`

File: client/components/web-preview/style.scss (hunk 3 above)
Comment (marekhrabe) [inline client/components/web-preview/style.scss]: Spot on! 👏 I actually used the variable but after that, I did some tweaking in devtools and copy&pasted the CSS back and thus lost the variable in favor of hex.

File: client/components/web-preview/style.scss (hunk 2 above)
Comment (marekhrabe) [inline client/components/web-preview/style.scss]: In general, you are totally right. In this case that would, however, mean that components using WebPreview that want to declare a loading message would need to use WebPreview internal classname in order to get a proper styling. Thus we opted to provide the option to use strong and we'll take care of styling internally.

An alternative to this approach (more correct but more constraining) would be to split the prop into two - something like `loadingMessageHeading` & `loadingMessageDescription` and then internally wrap them in elements with correct classnames.

**Comment by marekhrabe** [comment]
- [x] Rebased against my other updates in master
- [x] No more spinner
- [x] Safari & Firefox now center correctly

I'll leave the decision of loading message formulation & eventually the animation up to somebody else in Samus who will pick this up while I'm AFK. 

Just a quick note about the animation - we already have the SpinnerLine in place. The additional movement might be too much.

File: client/components/web-preview/style.scss (hunk 2 above)
Comment (spen) [inline client/components/web-preview/style.scss]: I figured there'd be good reason for it! Thanks for the info :)

**Comment by mcsf** [comment]
Can we add a comma at "One moment, please"? My second recommendation is to properly separate both fragments, either with a period or just with line breaking + capitalization:

- "One moment, please. Loading your new post."
- "One moment, please [line break] Loading your new post"

As for the language style behind "One moment, please", I don't have a formed opinion. I don't mind merging this with that sentence.

**Comment by marekhrabe** [comment]
@mcsf I like the second one more. The PR is all yours if you want — feel free to update, get reviewed & merge :)

**Comment by MichaelArestad** [comment]
Could we try a loading skeleton that represents a basic website layout rather than a grey loading screen?

**Comment by MichaelArestad** [comment]
Maybe something like this:

loading-site

**Comment by jblz** [comment]
@MichaelArestad In theory, I quite like that, but wouldn't it be kind of strange to show that & pop / fade in a site w/ a drastically different layout? I feel like even if the sidebar were on the opposite side, it'd feel odd.

🤔 I wonder if we can cache & prefetch some basic theme defaults and settings to make this a more realistic representation of the site's homepage in any meaningful way (that's worth the effort).

**Comment by MichaelArestad** [comment]
> @MichaelArestad In theory, I quite like that, but wouldn't it be kind of strange to show that & pop / fade in a site w/ a drastically different layout? I feel like even if the sidebar were on the opposite side, it'd feel odd.

I'm not sure. It's entirely possible it would feel strange. I don't know if it would be visible long enough to notice.

**Comment by spen** [comment]
I was thinking something similar to @jblz I think.
The skeleton state would be nice if it weren't so jarring for folks with different layouts (it might not be that bad, really)... Skeletons work best when the layout is predictable - Facebook, for instance, do a really good job of this but a lot of what makes it smooth is just how predictable the content would be.

My rough, not very well thought out solution might be to keep some very basic data in each theme that just says 'there's a medium header, a left hand side bar and a big content area' - but in a smart concise way. We could just use this to build the skeleton areas dead quick.
They'd have to be cached and available before the site loads of course.

Not sure if this is exactly what @jblz meant, so adding my £1.52 just in case.

**Comment by MichaelArestad** [comment]
@spen Yeah. I was toying with an idea similar, but I think it might be too complex and difficult to maintain for all of the themes. It may also be obsolete post-gutenberg release.

**Comment by spen** [comment]
> I think it might be too complex and difficult to maintain for all of the themes.
Agreed, it does sound like a maintainability nightmare waiting to happen.
I wonder if we could have a gif that looks like layout elements being moved around - I've got something in mind but it'd be hard to properly describe :/

**Comment by MichaelArestad** [comment]
Alternately, we could use on of our new illustrations. Perhaps something like this:

image

If none of them work super well, we could request a new one.

**Comment by spen** [comment]
Quite a fan of this illustration - though we'd still benefit from including something non-static (like the ellipsis originally mentioned) else it might feel as though nothing is happening in the background :)

**Comment by lancewillett** [comment]
Testing in Mac-n-Chrome, via Live Branch — looking good 👍 

load-site-new

To get this moving, could we: 1) merge with text + ellipsis now and 2) open a new issue to add the illustration once it's ready (later). Incremental changes FTW.

**Comment by lancewillett** [comment]
Testing note, small screen layout doesn't show the message:

load-site-new-phone

**Comment by spen** [comment]
Re: the suggestion in https://github.com/Automattic/wp-calypso/pull/16673#issuecomment-318738478
We discussed this and figured it might be difficult to apply across the board because of RTL languages and possibly other scenarios where an ellipsis doesn't make so much sense.

An option might be to shimmer or pulse the colour of the whole word subtly... I'll make sure I add this to a separate issue so that we can move this PR along :P

**Comment by lancewillett** [comment]
Cancel my above mobile / small screens comment. I wasn't testing on the live branch correctly. Looks good now.

**Pull request merged**